src/phylum/ci/ci_base.py

```diff
@@ -181,6 +184,30 @@ def depfiles(self) -> Depfiles:
             self.returncode = ReturnCode.NO_DEPFILES_PROVIDED
         raise SystemExit(self.returncode)
 
+    def _exclude_depfiles(self, provided_depfiles: DepfileEntries) -> DepfileEntries:
+        """Apply exclusion patterns to provided dependency files and return the remaining ones."""
+        arg_exclusions: Optional[list[list[str]]] = self.args.exclude
+        if not arg_exclusions:
+            LOG.debug("No dependency file exclusion patterns provided.")
+            return provided_depfiles
+
+        # flatten the list of lists
+        provided_arg_exclusions = list(chain.from_iterable(arg_exclusions))
+        LOG.debug("Exclusion patterns provided as arguments: %s", provided_arg_exclusions)
+
+        excluded_depfiles: DepfileEntries = []
+        for depfile in provided_depfiles:
+            for exclusion in provided_arg_exclusions:
+                if depfile.path.match(exclusion):
+                    excluded_depfiles.append(depfile)
+                    # Stop after first matching exclusion
+                    continue
```

Comment:
Does this actually stop? Wouldn't continue here just... continue? In any other language I'd expect a `break` for the inner loop here.

Reply:
Oops...you are correct. It should be `break`. I don't know what I was thinking. Change coming.